packages/sfui/frameworks/react/components/SfScrollable/SfScrollable.tsx

```diff
@@ -0,0 +1,126 @@
+import { useMemo } from 'react';
+import classNames from 'classnames';
+import {
+  polymorphicForwardRef,
+  SfButton,
+  SfIconChevronLeft,
+  SfIconChevronRight,
+  useScrollable,
+  SfScrollableDirection,
+  SfScrollableButtonsPlacement,
+  type SfScrollableProps,
+} from '@storefront-ui/react';
+
+const defaultScrollableTag = 'div';
+
+const SfScrollable = polymorphicForwardRef<typeof defaultScrollableTag, SfScrollableProps>(
+  (
+    {
+      as,
+      direction = SfScrollableDirection.horizontal,
+      buttonsPlacement = SfScrollableButtonsPlacement.blocked,
+      activeIndex,
+      reduceMotion,
+      drag,
+      onDragChange,
+      onScroll,
+      onPrev,
+      onNext,
+      className,
+      wrapperClassNames,
+      style,
+      children,
+      slotPreviousButton,
+      slotNextButton,
+      ...attributes
+    },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ref,
+  ) => {
+    const Tag = as || defaultScrollableTag;
+
+    const sliderOptions = useMemo(
+      () => ({
+        direction,
+        activeIndex,
+        reduceMotion,
+        drag,
+        onDragChange,
+        onScroll,
+        onPrev,
+        onNext,
+      }),
+      [direction, activeIndex, reduceMotion, drag, onDragChange, onScroll, onPrev, onNext],
+    );
+
+    const { getContainerProps, state, getNextButtonProps, getPrevButtonProps } = useScrollable(sliderOptions);
+
+    const previousButton = (...buttonClassName: Parameters<typeof classNames>) =>
+      (slotPreviousButton && <div {...getPrevButtonProps()}>{slotPreviousButton}</div>) || (
+        <SfButton
+          variant="secondary"
+          size="lg"
+          className={classNames(buttonClassName)}
+          square
+          slotPrefix={<SfIconChevronLeft />}
+          {...getPrevButtonProps()}
+        />
+      );
+
+    const nextButton = (...buttonClassName: Parameters<typeof classNames>) =>
+      (slotNextButton && <div {...getNextButtonProps()}>{slotNextButton}</div>) || (
+        <SfButton
+          variant="secondary"
+          size="lg"
+          square
+          className={classNames(buttonClassName)}
+          slotPrefix={<SfIconChevronRight />}
+          {...getNextButtonProps()}
+        />
+      );
+
+    const isHorizontal = direction === SfScrollableDirection.horizontal;
+
+    return (
+      <div
+        className={classNames(
+          'items-center relative',
+          isHorizontal ? 'flex' : 'flex-col h-full inline-flex',
+          wrapperClassNames,
```

Comment:
This is something I wasn't sure, which way go with component, its because how hook is build, but thats something we can change, I will think about it next week, even we will invert and use `{...attributes}` on root element and `innerElementAttr` this will be easy to migrate later on